lms/djangoapps/commerce/utils.py

```diff
@@ -109,3 +117,261 @@ def upgrade_url(self, user, course_key):
             else:
                 return reverse('verify_student_upgrade_and_verify', args=(course_key,))
         return None
+
+
+def refund_entitlement(course_entitlement):
+    """
+    Attempt a refund of a course entitlement. Verify the User before calling this refund method
+
+    Returns:
+        bool: True if the Refund is successfully processed.
+    """
+    user_model = get_user_model()
+    enrollee = course_entitlement.user
+    entitlement_uuid = str(course_entitlement.uuid)
+
+    if not is_commerce_service_configured():
+        log.error(
+            'Ecommerce service is not configured, cannot refund for user [%s], course entitlement [%s].',
+            enrollee.username,
+            entitlement_uuid
+        )
+        return False
+
+    service_user = user_model.objects.get(username=settings.ECOMMERCE_SERVICE_WORKER_USERNAME)
+    api_client = ecommerce_api_client(service_user)
+
+    log.info(
+        'Attempting to create a refund for user [%s], course entitlement [%s]...',
+        enrollee.username,
+        entitlement_uuid
+    )
+
+    try:
+        refund_ids = api_client.refunds.post(
+            {
+                'order_number': course_entitlement.order_number,
+                'username': enrollee.username,
+                'entitlement_uuid': entitlement_uuid,
+            }
+        )
+    except Exception as exc:  # pylint: disable=broad-except
+        # Catch any possible exceptions from the Ecommerce service to ensure we fail gracefully
+        log.exception(
+            "Unexpected exception while attempting to initiate refund for user [%s], "
+            "course entitlement [%s] message: [%s]",
+            course_entitlement.user.id,
+            course_entitlement.uuid,
+            str(exc)
+        )
+        return False
+
+    if refund_ids:
+        log.info(
+            'Refund successfully opened for user [%s], course entitlement [%s]: %r',
+            enrollee.username,
+            entitlement_uuid,
+            refund_ids,
+        )
+
+        return _process_refund(
+            refund_ids=refund_ids,
+            api_client=api_client,
+            course_product=course_entitlement
+        )
+    else:
+        log.warn('No refund opened for user [%s], course entitlement [%s]', enrollee.id, entitlement_uuid)
```

Comment:
Nit: all the other log messages use username, but this one uses user id. is there any particular reason for that?